core/src/main/java/org/infinispan/remoting/inboundhandler/DefaultTopologyRunnable.java

```diff
@@ -39,19 +38,31 @@ public boolean isReady() {
    }
 
    @Override
-   protected Response beforeInvoke() throws Exception {
+   protected InvocationStatus beforeInvoke() throws Exception {
+      // Read commands are executed directly in OOB thread pool. When the many commands are received prior
+      // to the first topology update (when the node is joining), these can wait for transaction data
+      // (even in non-tx mode) and deplete OOB TP => the node stops being responsive event to ST.
```

Comment:
You're right, in the bright future the command invocation will be asynchronous and we won't need `BlockingRunnable` and its subclasses at all. However, I don't expect to be able to remove `canBlock()` in version 9.0.